pages/index.tsx

```diff
@@ -0,0 +1,30 @@
+import styles from '../styles/Home.module.css'
+import { getHeroUnit } from "../lib/kontentClient";
+import { GetStaticProps, NextPage } from 'next';
+import { HeroUnit } from '../models/content-types/hero_unit';
+
+const Home: NextPage<IndexProps> = ({ heroUnit }) => {
+  return (
+    <main >
+      <div className={styles.hero}>
+        <h1 className="append-dot">{heroUnit.elements.title.value}</h1>
+        <div className={styles.summary} dangerouslySetInnerHTML={{ __html: heroUnit.elements.marketingMessage.value }}>
+        </div>
+      </div>
+    </main>
+  )
+}
+
+export default Home
+
+interface IndexProps {
+  heroUnit: HeroUnit;
+}
+
+export const getStaticProps: GetStaticProps<IndexProps> = async () => {
```

Comment:
You had typed return value in client, I think it also belongs here :) I think, that eslint can be setup to control that, however not sure whether it is suitable for templates. For me, I don't see a reason why not. What do you think? :)

Reply:
The return type in client return `HeroUnit` -> this index defines a shape of the props being passed -> and object with a property `heroUnit` of the`Herounit` type. 

I would say it helps you to return always the consistent shape of data you expect in props -> it is some kind of a link between server data shape and props shape that goes to the client.

I have set TS config to strict mode as well as selling configuration and set up the CI to have this check in place.